test/jdk/java/nio/channels/Channels/TransferTo.java

```diff
@@ -282,4 +280,29 @@ public OutputStream output(Consumer<Supplier<byte[]>> spy) throws Exception {
         };
     }
 
+    @BeforeClass
+    public static void createTempFileRegistry() {
+        tempFiles = new LinkedList();
+    }
+
+    @AfterClass
+    public static void deleteTempFiles() {
+        tempFiles.forEach(path -> {
+            try {
+                Files.delete(path);
+            } catch (IOException e) {
+                // ignored
+            }
+        });
+        tempFiles = null;
+    }
+
+    /*
+     * Creates a temporary file and registers it for later cleanup after tests.
+     */
+    private static Path newTempFile() throws IOException {
```

Comment:
You might consider passing in the prefix to make it easier to differentiate the temp file (InputStreamProvider vs OutputSteamProvider...)

Reply:
Is this really worth the extra complexity? What is the use case where this plays a vital role?